python/tank/platform/qt/tankqdialog.py

```diff
@@ -299,10 +299,10 @@ def _format_context_property(p, show_type=False):
                 if p is None:
                     formatted = "Undefined"
                 elif show_type:
-                    formatted = "%s %s" % (six.ensure_str(p.get("type")),
-                                           six.ensure_str(p.get("name")))
+                    formatted = "%s %s" % (six.ensure_str(p.get("type", "None")),
+                                           six.ensure_str(p.get("name", "None")))
                 else:
-                    formatted = "%s" % six.ensure_str(p.get("name"))
+                    formatted = "%s" % six.ensure_str(p.get("name", "None"))
```

Comment:
`"None"` or `None`?

Reply:
Before `ensure_str` was used, `p.get` returns None object by default. The string interpolation printed the word None. 

Now that we are using `six.ensure_str`, it requires a string argument otherwise it breaks.

It should be a string with the word `None` to keep it backwards compatible and make sure we don't pass a None object to `ensure_str`. We can use empty string too but the tooltip will look weird.

Reply:
I think that's why my recommendation in #865 was to remove the call to `six.ensure_str`. I did dig into where it is coming from and I remember it being OK.
Also now we don't support Py2 anymore.

But I won't make this a blocker so you decide what is the best in this situation.

Reply:
If removing `six.ensure_str` is safe at this point, I also think it's better to get rid of it. Thanks for sharing the context.